bundles/core/src/main/java/com/adobe/cq/wcm/core/components/util/ComponentUtils.java

```diff
@@ -170,4 +212,88 @@ private static String generateId(@NotNull final Resource resource,
+		
+		String styleClasses = null;
+		
+		if (componentResource != null) {
+			ResourceResolver resourceResolver = componentResource.getResourceResolver();
+			ValueMap valueMap = componentResource.getValueMap();
+			String[] styleids = valueMap.get(CQ_STYLEIDS, String[].class);
+			if (styleids != null && ArrayUtils.isNotEmpty(styleids)) {
+				ContentPolicyManager policyMgr = resourceResolver.adaptTo(ContentPolicyManager.class);
+				if (policyMgr != null) {
+
+					ContentPolicy contentPolicy = policyMgr.getPolicy(componentResource);
+					if (contentPolicy != null) {
+						
+						String defaultStyle = (String) contentPolicy.getProperties().get(CQ_STYLECLASSES_DEFAULT);
+
+						styleClasses = StringUtils.defaultIfBlank(String.join(StringUtils.SPACE, getStyleSystemClassesFromIds(contentPolicy, styleids)),defaultStyle);
+					}
+				}
+			}
+		}
+		
+		return styleClasses;
+	}
+	
+	/**
+	 * @param contentPolicy
+	 * @param styleids
+	 * @return Return equivalent CSS classes separated using an empty space between each, for the list of style ids shared against the content policy to look for
+	 */
+	public static ArrayList<String> getStyleSystemClassesFromIds(ContentPolicy contentPolicy, String[] styleids) {
+		ArrayList<String> styleClasses = new ArrayList<String>();
+			if (styleids != null && ArrayUtils.isNotEmpty(styleids)) {
+				HashMap<String, String> styleMap = new HashMap<String, String>();
+
+					if (contentPolicy != null) {
+
+						Resource contentPolicyResource = contentPolicy.adaptTo(Resource.class);
+						if(contentPolicyResource != null) {
+							Resource styleGroupsResource = contentPolicyResource.getChild(
+									CQ_STYLEGROUPS);
+							if (styleGroupsResource != null) {
+								Iterator<Resource> styleGroupsIterator = styleGroupsResource.listChildren();
+								while (styleGroupsIterator.hasNext()) {
+									Resource styleGroupItem = styleGroupsIterator.next();
+									Resource cqStylesResource = styleGroupItem.getChild(CQ_STYLES);
+									if(cqStylesResource != null)
+									{
+										Iterator<Resource> cqStylesIterator = cqStylesResource.listChildren();
+										while (cqStylesIterator.hasNext()) {
+											Resource cqStyleDefinition = cqStylesIterator.next();
+											ValueMap childProperties = cqStyleDefinition.getValueMap();
+											String key = childProperties.get(CQ_STYLEID, String.class);
+											String value = childProperties.get(CQ_STYLECLASSES,
+													StringUtils.EMPTY);
+											if (StringUtils.isNotBlank(key)) {
+												styleMap.put(key, value);
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				
+				
+				  for (Entry<String, String> entry : styleMap.entrySet()) {
+					    String styleid = entry.getKey();
+					    String styleClass = entry.getValue();
+					    if (ArrayUtils.contains(styleids, styleid)) {
+					    	styleClasses.add(styleClass);
+						}
+					  }
+			}
+		
+		return styleClasses;
+	}
```

Comment:
This should leverage the implementation of `ComponentStyleInfo#getAppliedStyles` instead of reimplementing everything.

Reply:
Can you please share the GitHub URL for this class? I wasn't able to locate. If its part of a different bundle or is-a different service, please advise if I can do the required integration (Sorry I didn't ever come across this class). Thank you !